src/app/profile/profile.component.js

```diff
@@ -0,0 +1,385 @@
+            this.updateEmail(true);
+          }
+        }
+      );
+  }
+
+  loadEmail() {
+    this.emailLoading = true;
+    this.profileService.getEmails()
+      .subscribe(
+        emails => {
+          this.donorEmail = emails ? emails[0] : '';
+          this.spouseEmail = emails ? emails[1] : '';
+          this.emailLoading = false;
+        },
+        error => {
+          this.emailAddressError = 'Failed loading email.';
+          this.emailLoading = false;
+          this.$log.error(this.emailAddressError, error.data);
+        }
+      );
+  }
+
+  updateEmail(spouse) {
+    let email =  spouse ? this.spouseEmail : this.donorEmail;
+    this.emailLoading = true;
+    this.profileService.updateEmail(email,spouse)
+      .subscribe(
+        data => {
+          if(spouse) {
+            this.spouseEmail = data;
+            this.spouseEmailForm.$setPristine();
+          } else {
+            this.donorEmail = data;
+            this.donorEmailForm.$setPristine();
+          }
+          this.success = true;
+          this.emailLoading = false;
+        },
+        error => {
+          this.emailAddressError = 'Failed updating email address.';
+          this.$log.error(this.emailAddressError, error);
+          this.emailLoading = false;
+        }
+      );
+  }
+
+  loadPhoneNumbers() {
+    this.phonesLoading = true;
+    this.profileService.getPhoneNumbers()
+      .subscribe(
+        data => {
+          this.phonesLoading = false;
+          angular.forEach(data, (item) => {
+            item.ownerChanged = false;
+            this.phoneNumbers.push(item);
+          });
+        },
+        error => {
+          this.phoneNumberError = 'Failed loading phone numbers.';
+          this.$log.error(this.phoneNumberError, error.data);
+          this.phonesLoading = false;
+        }
+      );
+  }
+
+  addPhoneNumber() {
+    this.phoneNumbers.push({
+      'phone-number': '',
+      'phone-number-type': 'Mobile',
+      'primary': false,
+      'spouse': false
+    });
+  }
+
+  phoneOwner(spouse) {
+    return spouse ? this.donorDetails['spouse-name']['given-name'] : this.donorDetails.name['given-name'];
+  }
+
+  updatePhoneNumbers(){
+    for(let i=0;i<this.phoneNumbers.length;i++) {
+      let item = this.phoneNumbers[i];
+      this.phonesLoading = true;
+
+      if(item.ownerChanged) { // switch of phone owner (set to delete phone and add it again with new owner)
+        item.delete = true;
+        this.phoneNumbers.push({
+          'phone-number': item['phone-number'],
+          'phone-number-type': item['phone-number-type'],
+          'primary': false,
+          'spouse': item.spouse
+        });
+      }
+      if(item.self && item.delete == undefined) { // update existing phone number
+        this.profileService.updatePhoneNumber(item)
+          .subscribe(
+            () => {
+              this.resetPhoneNumberForms();
+              this.success = true;
```

Comment:
Seems like the first request that returns is going to set success to true and call reset `PhoneNumberForms`, regardless if the other requests have finished or failed. Idk if that is an issue.
You could build up an array of requests and send them off at once with RxJS's forkJoin. That way you'd have only one subscribe that would call the success function only once they have all succeeded or the error function if any of them failed. Handling some successes and but some failures is more complicated. Idk if this context warrants that. You could take a look at https://github.com/CruGlobal/give-web/blob/master/src/app/profile/yourGiving/editRecurringGifts/step4/confirmRecurringGifts.component.js#L37-L64